File: problems/1_Two_Sum/1.two-sum.java
import java.util.HashMap;

class Solution {
	public int[] twoSum(int[] nums, int target) {
		var map = new HashMap<Integer, Integer>();

		for (int i = 0; i < nums.length; i++) {
			map.put(target - nums[i], i);
		}

		for (int i = 0; i < nums.length; i++) {
			if (map.containsKey(nums[i]) && i != map.get(nums[i])) {
				return new int[] {i, map.get(nums[i])};
			}
		}

		return new int[] {};
	}
}

File: problems/1_Two_Sum/1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */
#include <unordered_map>

// @lc code=start
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        std::unordered_map<int, int> map{};
        for (int i = 0; i < nums.size(); ++i)
        {
            int complement = target - nums[i];
            if (map.count(complement))
            {
                return {map.at(complement), i};
            }
            map.emplace(nums[i], i);
        }
        return {};
    }
};
// @lc code=end
